File: Day-36.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int goodNodes(TreeNode root) {
        if(root == null) return 0;
        return dfs(root, root.val);
    }
    int dfs(TreeNode root, int max) {
        if(root == null) return 0;
        max = Math.max(root.val , max);
        if(root.val>=max)
            return 1+ dfs(root.left,max) + dfs(root.right,max);
        else
            return dfs(root.left,max) + dfs(root.right,max);
    }
}

File: Day-36.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
 const dfs = (root, arr) => {
     if(!root) return ;
     if(!root.left && !root.right){
          arr.push(root.val);
          return;
     }
     dfs(root.left, arr)
     dfs(root.right, arr)

 }
var leafSimilar = function(root1, root2) {
    let arr1 = [];
    let arr2 = [];
      dfs(root1, arr1)
      dfs(root2, arr2)
    return String(arr1) === String(arr2)
};
